static/app/components/core/compactSelect/control.tsx

```diff
@@ -654,39 +655,23 @@ const ClearButton = styled(Button)`
   margin: -${space(0.25)} -${space(0.5)};
 `;
 
-const searchVerticalPadding: Record<FormSize, string> = {
-  xs: space(0.25),
-  sm: space(0.5),
-  md: space(0.5),
+const searchInputSizeMapping: Record<FormSize, FormSize> = {
+  md: 'sm',
+  sm: 'xs',
+  xs: 'xs',
 };
```

Comment:
@Jesse-Box I had to make the search input smaller compared to the size the actual `compactSelect` gets passed. So when the select is `md`, the search input is `sm` etc.

But for `xs`, there is no smaller version to go towards, so I just kept it `xs`